src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -67,7 +68,10 @@ function DimensionMapper(props: Props): JSX.Element {
             const newDim = Number(val);
             if (selectedDim !== newDim) {
               const newMapperState = mapperState.slice();
-              newMapperState[selectedDim] = 0; // reset slicing index of previously selected dimension
+              // Invert mappings or reset slicing index of previously selected dimension
```

Comment:
I invert the mapping of `selectedDim` and `newDim` and reset the slicing index to 0 only if it was neither `x` nor `y` (i.e. a number).